packages/rest/src/rest.server.ts

```diff
@@ -920,11 +930,6 @@ export interface OpenApiSpecForm {
   format?: string;
 }
 
-const OPENAPI_SPEC_MAPPING: {[key: string]: OpenApiSpecForm} = {
-  '/openapi.json': {version: '3.0.0', format: 'json'},
```

Comment:
IMO, we should preserve `OPENAPI_SPEC_MAPPING` as a constant config independent on `basePath`. The `basePath` setting should be applied when we are processing the mapping entries.

In your proposal, when app developers specify custom mapping, then their custom endpoints won't be prefixed with `basePath`. I would find that surprising.

In my proposal, `basePath` will be prepended to all mapping entries, regardless of whether they are provided by the user or come from the default mapping.

Reply:
Leaving OPENAPI_SPEC_MAPPING where it is and applying `basePath` to it inside the RestServer constructor will mean recreating the entire OPENAPI_SPEC_MAPPING object with `basePath` added to the key names. 
Is there a reason to do it considering that  OPENAPI_SPEC_MAPPING constant isn't used anywhere else in the code other than RestServer constructor?

Yes, I propose that custom mapping should be truly custom and not be altered once it is set by a developer. IMO this will give developers more flexibility. But I'm fine either way as long as we have this PR moving forward. 

My main blocker is inability to customize `/explorer` path.

Reply:
> My main blocker is inability to customize /explorer path.

Yes, that's what I'm seeing now. Please `@loopback/rest-explorer` mounts the UI as follows:

```ts
application.static(explorerPath, swaggerUI.getAbsoluteFSPath()); // explorePath is default to `/explorer`
```

But `RestApplication.static()` adds it to a sub router to the root router (/api). As a result, the UI is only available under `/api/explorer` while the spec is served at `/openapi.json`. Here is the express pipeline:

```
/ (RestServer._expressApp)
  /openapi.json
  /openapi.yaml
  /api
    /products
    /explorer (Explorer UI) 
```

If we want keep `/openapi.json` and `/explorer` regardless of `basePath`, we need to allow `/explorer` to be mounted at root `RestServer._expressApp` level.

Reply:
> Leaving OPENAPI_SPEC_MAPPING where it is and applying basePath to it inside the RestServer constructor will mean recreating the entire OPENAPI_SPEC_MAPPING object with basePath added to the key names.

I am envisioning this differently. Instead of re-creating a new mapping object, we should apply `basePath` when creating Express routes here:

https://github.com/strongloop/loopback-next/blob/6e0eeb66b6a8c79d736fcc2dc7d9c76bfd57abd3/packages/rest/src/rest.server.ts#L281-L286

```diff
-    this._expressApp.get(p, (req, res) => 
+    this._expressApp.get(basePath + p, (req, res) => 
```